mt-kahypar/partition/initial_partitioning/initial_partitioning_commons.h

```diff
@@ -31,15 +31,25 @@
 #include "kahypar-resources/datastructure/kway_priority_queue.h"
 #pragma GCC diagnostic pop
 
+#include "kahypar-resources/datastructure/fast_reset_flag_array.h"
+#include "kahypar-resources/meta/abstract_factory.h"
+
 #include "tbb/enumerable_thread_specific.h"
+
 #include "mt-kahypar/datastructures/hypergraph_common.h"
-#include "kahypar-resources/datastructure/fast_reset_flag_array.h"
+#include "mt-kahypar/definitions.h"
+#include "mt-kahypar/partition/initial_partitioning/i_initial_partitioner.h"
 
 namespace mt_kahypar {
 
+typedef struct ip_data_container_s ip_data_container_t;
+
 using KWayPriorityQueue = kahypar::ds::KWayPriorityQueue<HypernodeID, Gain, std::numeric_limits<Gain>, false>;
 using ThreadLocalKWayPriorityQueue = tbb::enumerable_thread_specific<KWayPriorityQueue>;
 
 using ThreadLocalFastResetFlagArray = tbb::enumerable_thread_specific<kahypar::ds::FastResetFlagArray<> >;
 
+using InitialPartitionerFactory = kahypar::meta::Factory<InitialPartitioningAlgorithm,
```

Comment:
Not sure what the best policy is, but let's discuss. Factories close to the definitions or factories in a dedicated factory header? We should try to be consistent, i.e., the refiner and coarsener factories are in a dedicated header.

Reply:
True. I guess it would make sense to move `InitialPartitionerFactory` also to `factories.h`